test/e2e/testsuites/testsuites.go

```diff
@@ -1046,6 +1071,61 @@ func SetNodeTaints(c clientset.Interface, node *v1.Node, taints ...v1.Taint) (cl
 	return
 }
 
+func AnnotateNode(c clientset.Interface, node *v1.Node, newAnnotations map[string]string) (newNode *v1.Node, cleanup func(), err error) {
+	originalAnnotations := node.Annotations
+	nodeName := node.Name
+	cleanup = func() {
+		node, err := c.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		node.Annotations = originalAnnotations
+		_, _ = c.CoreV1().Nodes().Update(context.Background(), node, metav1.UpdateOptions{})
+	}
+
+	if node.Annotations == nil {
+		node.Annotations = newAnnotations
+	} else {
+		for key, value := range newAnnotations {
+			node.Annotations[key] = value
+		}
+	}
+
+	newNode, err = c.CoreV1().Nodes().Update(context.Background(), node, metav1.UpdateOptions{})
+	if err != nil {
+		klog.Errorf("failed to add annotations (%+v) to node (%s): %v", newAnnotations, node.Name, err)
+	} else {
+		klog.Infof("successfully added annotations (%+v) to node (%s)", newAnnotations, node.Name)
+	}
+	return
+}
+
+func SetCSINodeDriver(c clientset.Interface, nodeName string) (cleanup func(), err error) {
+	csiNode, err := c.StorageV1().CSINodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	originalSpec := csiNode.Spec
+	cleanup = func() {
+		csiNode, err := c.StorageV1().CSINodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		csiNode.Spec = originalSpec
+		_, _ = c.StorageV1().CSINodes().Update(context.Background(), csiNode, metav1.UpdateOptions{})
+	}
+
+	if csiNode.Spec.Drivers == nil || len(csiNode.Spec.Drivers) == 0 {
+		csiNode.Spec.Drivers = []storagev1.CSINodeDriver{
+			{
+				Name:         "secrets-store.csi.k8s.io",
```

Comment:
Why do we need to add this driver to the node spec?

Reply:
This was required to schedule a pod onto master node, otherwise, external-provisioner complains that `no topology key was found in the csi node`. 

https://github.com/kubernetes-csi/external-provisioner/blob/7ae61466a611061ba70da2679358c8bfc8546b6f/pkg/controller/topology.go#L114-L134